### ISSUE TITLE
Incorrect shop backend API request during renewal purchase process.

### ISSUE THREAD

**Issue by cmars** [canonical-web-and-design/ubuntu.com#9107]
*Note - filling in this form will file an issue against the www.ubuntu.com website, not against Ubuntu itself, for Ubuntu bugs, please [go here](https://help.ubuntu.com/community/ReportingBugs)*

## Summary

Ubuntu Advantage shop appears to be making an incorrect API request during renewal purchase process.

## Process

This appears to be caused by a customer purchasing a renewal. We see an API call to `/v1/renewals/rREDACTED/purchase/preview`.

## Current and expected result

Purchase preview is only supported by the backend for new purchases. It doesn't work with old-style renewals yet (which pre-date the shop). So this API endpoint shouldn't be called for renewals.

## Screenshot

Not available for data protection reasons.

## Browser details

We don't have this information, but if you ping us in chat, we might be able to get customer success to get this information.

**Comment by cmars** [comment]
FWIW I don't think it's a good idea to use a public Github repository for shop issues. A private project would be more appropriate. Or Launchpad, if we need a mix of public and private issues.

**Comment by anthonydillon** [comment]
Thanks @cmars. If there is something truly private then we should use a more private channel. But, in this case I don't see anything specifically private.

**Comment by alnvdl** [comment]
Just to add a bit more of context to this: the [renewal preview endpoint](https://github.com/CanonicalLtd/ua-contracts/blob/3b843a7f2ffe4bfc3e62d965ecc77c286d23113c/docs/contracts.yaml#L4290) can only be used if the renewal is being accepted as if it were purchases, which has to be done via [a dedicated endpoint](https://github.com/CanonicalLtd/ua-contracts/blob/3b843a7f2ffe4bfc3e62d965ecc77c286d23113c/docs/contracts.yaml#L4327). It also means the renewal ends up in the same subscription as another shop purchase would.

I think there was a miscommunication, because only the first part of it is being performed (the preview part). I will update the API doc to make that clear.

For now, it's best to stick [to the current endpoint the web interface is using to accept renewals](https://github.com/CanonicalLtd/ua-contracts/blob/3b843a7f2ffe4bfc3e62d965ecc77c286d23113c/docs/contracts.yaml#L4253) and remove the estimation feature, as it would not be accurate at all (if the customer has an existing shop subscription, it would show a misleading value).

Once the migration to the new endpoints is done, it would be necessary to retest renewals from scratch. Given all the other tasks, I'd say this is probably lower priority.
